session.go

```diff
@@ -1,97 +1,77 @@
 package bond
 
 import (
+	"database/sql"
+	"fmt"
 	"reflect"
 	"sync"
 
-	"upper.io/db"
+	"upper.io/db.v2"
+	"upper.io/db.v2/lib"
 )
 
 type Session interface {
-	db.Tx
+	lib.SQLDatabase
```

Comment:
The main problem with allowing Session to have Commit/Rollback is that then session becomes aware of the transaction, we'll have sessions with transactions and sessions without them. This lead us to have odd things like ContinueTransaction. My idea here was making bond not aware of the transaction and force that logic out, let the developer deal with the transaction and create new ones, use the same one, commit it, etc. The only place where bond has a transaction is on the BondTx() method, and in this case the transactions is always commited/rolledback at the end.